File: Data Structure&Algorithm/meeting/xiecheng/1demo.py
def main():
    arr=[]
    res=[]
    a=int(input())
    b=int(input())
    k=int(input())
    arr.append(a)
    arr.append(b)
    arr.sort()
    for i in range(k,-1,-1):
        res.append(i*arr[0]+(k-i)*arr[1])
    res=list(set(res))
    res.sort()
    if res==[0]:
        print([])
    else:
        print(res)
main()
# def dfs(k,ans):
#     if k==0:
#         res.append(ans)
#         return
#     for item in arr:
#         dfs(k-1,ans+item)

File: Data Structure&Algorithm/meeting/xiecheng/1demo.cpp
#include<iostream>
#include<vector>
#include<map>
using namespace std;
vector<int> arr;
vector<int> res;
map<int,bool> mm;
void dfs(int k,int ans)
{
    if(k==0&&!mm[ans]){
        mm[ans]=true;
        res.push_back(ans);
        return;
    }
    for(int i=0;i<arr.size();i++){
        dfs(k-1,ans+arr[i]);
    }
}
int main(){
    int a,b,k;
    cin>>a>>b>>k;
    arr.push_back(a);
    arr.push_back(b);
    dfs(k,0);
}